### ISSUE TITLE
Pluginn not connecting to kite engine

### ISSUE THREAD

**Issue by tooqui** [kiteco/atom-plugin#450]
Hi All,
I am trying to start using kite with atom but for some reason I can not make it work. 
I am using a MAC latest sierra release. 
Insatalled latest kite engine version (also signed up and configured)
Installed latest atom and plugin from Setting/install  (also tried the command line install) 
No matter what, the plugin does not seem to notice the engine is running on the computer..

Any ideas?
![atom](https://user-images.githubusercontent.com/4553065/42041170-a140889e-7af1-11e8-9a2e-b5de51f95272.jpeg)

**Comment by abe33** [comment]
Hi @tooqui, thanks for the report

If you don't mind, can you try a few thing to help gather some information about the issue? 

- What is the error message you get in Atom? is it kite not installed, or kite not running (if you're not sure how to find it you can click on the kite icon in the status bar to open the status panel)
- Are you seeing the Kite icon in the menu bar? If not, are you seeing Kite.app process when running `ps -x | grep Kite` in a terminal?

**Comment by abe33** [comment]
Thanks for the quick feedback, I'm looking into this right now, I'll come back as soon as I have some idea of what's happening

**Comment by tooqui** [comment]
Hi Abe33,
Thank YOU for the support. Here is the information you requested.
![atom](https://user-images.githubusercontent.com/4553065/42042276-14e48ac8-7af4-11e8-9fd4-fb519303dee8.jpeg)

I also opened http://localhost:46624/settings and can see the Kite information through the browser...  to me it seems the plugin does not speak with engine for some reason...

OS firewall is disabled. Was first I tried :)

**Comment by abe33** [comment]
@tooqui thanks for the quick reply, I think the issue comes from the fact that Kite 0.1.0 is a really old version, I'm not sure how you got this installed, but that's definitely the main issue here (we're using a 0.<release-date>.0 scheme now, and latest is 0.20180627.0 currently). I would advise trying to uninstall your current version and download a fresh one from the website. 

Maybe a last question if you don't mind, how did you download Kite to get this old version, was that you downloaded it a long time ago and only installed it recently or is it an issue on our that we should fix (maybe something went wrong with our website, I'd rather make sure everything is ok)

**Comment by abe33** [comment]
@tooqui if you don't mind, can you try this in a terminal: `mdfind kMDItemCFBundleIdentifier = "com.kite.Kite"` that's normally what we're using to check whether kite is installed or not in the plugin

**Comment by tooqui** [comment]
Hi Abe33,
I had kite installed but never actually used it.  Now I decided to use it so deleted kite from applications folder and downloaded kite from kite project. the file name is Kite-0.20180627.0.dmg
I am having issues unistalling that version...  
I am deleting my /%HOME%/.kite  folder with [sudo rm -r .kite]
and deleting the app from the applications folder. 

Tried to reinstall and no matter what I do come back with version 0.1.0


Any ideas beside setting the computer on fire? hhhhhh  I have the matches on hand already...

**Comment by tooqui** [comment]
mdfind kMDItemCFBundleIdentifier = "com.kite.Kite"
shows nothing

MacBook-Pro-de-Jay:~ jay$ mdfind kMDItemCFBundleIdentifier = "com.kite.Kite"
MacBook-Pro-de-Jay:~ jay$

**Comment by tooqui** [comment]
Actually, I can see the About kite menu option brings up that 0.1.0 letter but bottom of the actual kite app I can see a different version (current one)

![atom](https://user-images.githubusercontent.com/4553065/42044850-591d587c-7afa-11e8-8bc8-87694a830ed6.jpeg)

**Comment by abe33** [comment]
@tooqui
> Any ideas beside setting the computer on fire? hhhhhh I have the matches on hand already...
 haha, no need to go that far

> Actually, I can see the About kite menu option brings up that 0.1.0 letter but bottom of the actual kite app I can see a different version (current one)

Yeah, I was confused by that as well, the about command of the copilot open the electron app about page, and the version displayed here is the one from the electron app. It's definitely confusing so I'll report that to my team mates in charge of the copilot.

> mdfind kMDItemCFBundleIdentifier = "com.kite.Kite" shows nothing
Ok, that's definitely the issue here, we're using that command to find the kite install path and make sure kite is installed. I'm not an OSX specialist so I'll have to ask for help and I'll come back at you as soon as I have more information/question about that issue.

**Comment by tooqui** [comment]
@abe33 thank you for your help.  Will wait for your info and keep trying to address the issue.

**Comment by tooqui** [comment]
Just if it is of any help.
on /Library/Preferences/
`MacBook-Pro-de-Jacobo:Preferences jacobourdiales$ ls | grep kite`
`com.kite.Kite.plist`
`com.kite.KiteApp.helper.plist`
`com.kite.KiteApp.plist`
`com.kite.KiteSidebar.plist`

**Comment by abe33** [comment]
@tooqui, I got some more information, apparently it seems like an issue with spotlight indexing, here's a few things: 
- is spotlight enabled? (if not `mdfind` won't work here)
- if spotlight is enabled or didn't caught up even after start, you can manually add Kite using `mdimport /Applications/Kite.app`. With this command mdfind should now return something for `mdfind kMDItemCFBundleIdentifier = "com.kite.Kite"`

Also, as a side note, we have a version in a PR that will get rid of these checks before requests to speed and make the plugin more reliable.

Hope it helps, feel free to ask if you get any other issue with kite.

**Comment by tooqui** [comment]
@abe33, Situation is spotlight is enabled and can not be disabled since OSX has a system integrity protection that does not allow you to disable it.  As these kind of programs take a lot of resources I added my full HD to the black list so actually no indexing is taking place.

I tried to enter the commands you stated on the terminal but there is no result.
`MacBook-Pro-de-Jay:~ jay$ mdimport /Applications/Kite.app`
`MacBook-Pro-de-Jay:~ jay$ mdfind kMDItemCFBundleIdentifier = "com.kite.Kite"`
`MacBook-Pro-de-Jay:~ jay$ `

**I confirm that when I removed the HD from the spotlight blacklist the sync started to work fine.
Thank you very much for your help!**
